test/checks/tables/th-has-data-cells.js

```diff
@@ -24,6 +24,8 @@ describe('th-has-data-cells', function () {
 			'<table>' +
 			'  <tr> <th>hi</th> <td>hello</td> </tr>' +
 			'  <tr> <th>hi</th> <td>hello</td> </tr>' +
+			'  <tr> <td>hello</td> <th>hi</th> </tr>' +
```

Comment:
I agree that support for this markup is poor across browser screen reader combinations. I tested on Windows 10 using IE11/JAWS 17, IE11/NVDA 2017.1, Firefox/JAWS 17, Firefox/NVDA 2017.1, Chrome/JAWS 17, and Chrome/NVDA 2017.1 and only Chrome/NVDA 2017.1 read the right-aligned row headers.

On macOS, the right-aligned cells were announced with Chrome/VoiceOver, but not with Safari/VoiceOver.

The right-aligned cells were not announced with iOS/VoiceOver or either Chrome or Firefox on Android with Talkback.

The use case where I've seen this happening involves repetition of column headers in a TFOOT, or a row header repeated as the last cell in a row in long and wide data tables. So it's not like the cells aren't labelled, but I'm not sure if the test should account for this or not. Also a row header in the middle of the row would pass this test as originally written because we are just checking to make sure that at least one subsequent data cell in the same row as the TH has content.